gk/fib.c

```diff
@@ -1664,3 +1666,407 @@ del_fib_entry(const char *ip_prefix, struct gk_config *gk_conf)
+	}
+
+	/* First argument must be of type CTYPE_STRUCT_GK_CONFIG_PTR. */
+	luaL_checkcdata(l, 1, &ctypeid, CTYPE_STRUCT_GK_CONFIG_PTR);
+	if (ctypeid != correct_ctypeid_gk_config)
+		luaL_error(l, "Expected `%s' as first argument",
+			CTYPE_STRUCT_GK_CONFIG_PTR);
+
+	/* Second argument must be a Lua function. */
+	luaL_checktype(l, 2, LUA_TFUNCTION);
+
+	/* Second argument must be a Lua string. */
+	luaL_checktype(l, 3, LUA_TSTRING);
+
+	gk_conf = *(struct gk_config **)
+		luaL_checkcdata(l, 1, &ctypeid, CTYPE_STRUCT_GK_CONFIG_PTR);
+
+	lua_remove(l, 1);
+
+	ltbl = &gk_conf->lpm_tbl;
+
+	rte_spinlock_lock_tm(&gk_conf->lpm_tbl.lock);
+	list_ipv6_fib_entries(l, ltbl);
+	rte_spinlock_unlock_tm(&gk_conf->lpm_tbl.lock);
+
+	lua_remove(l, 1);
+
+	return 1;
+}
+
+#define CTYPE_STRUCT_ETHER_ADDR_PTR "struct ether_addr *"
+
+static int
+l_ether_format_addr(lua_State *l)
+{
+	static bool assigned_type_ether_addr = false;
+	static uint32_t correct_ctypeid_ether_addr;
+
+	uint32_t ctypeid;
+	struct ether_addr *d_addr;
+	char d_buf[ETHER_ADDR_FMT_SIZE];
+
+	if (!assigned_type_ether_addr) {
+		correct_ctypeid_ether_addr = luaL_get_ctypeid(l,
+			CTYPE_STRUCT_ETHER_ADDR_PTR);
+		assigned_type_ether_addr = true;
+	}
+
+	/* First argument must be of type CTYPE_STRUCT_ETHER_ADDR_PTR. */
+	luaL_checkcdata(l, 1, &ctypeid, CTYPE_STRUCT_ETHER_ADDR_PTR);
+	if (ctypeid != correct_ctypeid_ether_addr)
+		luaL_error(l, "Expected `%s' as first argument",
+			CTYPE_STRUCT_ETHER_ADDR_PTR);
+
+	d_addr = *(struct ether_addr **)
+		luaL_checkcdata(l, 1, &ctypeid, CTYPE_STRUCT_ETHER_ADDR_PTR);
+
+	ether_format_addr(d_buf, sizeof(d_buf), d_addr);
+
+	lua_pushstring(l, d_buf);
+
+	return 1;
+}
+
+#define CTYPE_STRUCT_IP_ADDR_PTR "struct ipaddr *"
+
+static int
+l_ip_format_addr(lua_State *l)
+{
+	int ret;
+	static bool assigned_type_ip_addr = false;
+	static uint32_t correct_ctypeid_ip_addr;
+
+	uint32_t ctypeid;
+	struct ipaddr *ip_addr;
+	char ip[128];
+
+	if (!assigned_type_ip_addr) {
+		correct_ctypeid_ip_addr = luaL_get_ctypeid(l,
+			CTYPE_STRUCT_IP_ADDR_PTR);
+		assigned_type_ip_addr = true;
+	}
+
+	/* First argument must be of type CTYPE_STRUCT_IP_ADDR_PTR. */
+	luaL_checkcdata(l, 1, &ctypeid, CTYPE_STRUCT_IP_ADDR_PTR);
+	if (ctypeid != correct_ctypeid_ip_addr)
+		luaL_error(l, "Expected `%s' as first argument",
+			CTYPE_STRUCT_IP_ADDR_PTR);
+
+	ip_addr = *(struct ipaddr **)
+		luaL_checkcdata(l, 1, &ctypeid, CTYPE_STRUCT_IP_ADDR_PTR);
+
+	ret = convert_ip_to_str(ip_addr, ip, sizeof(ip));
+	if (ret < 0)
+		luaL_error(l, "gk: failed to convert an IP address to string!");
+
+	lua_pushstring(l, ip);
+
+	return 1;
+}
+
+static const struct luaL_reg dylib [] = {
+	{"list_fib4_entries", l_list_fib4_entries},
+	{"list_fib6_entries", l_list_fib6_entries},
+	{"ether_format_addr", l_ether_format_addr},
+	{"ip_format_addr",    l_ip_format_addr},
+	{NULL,                NULL}	/* Sentinel. */
+};
+
+const luaL_Reg *
+get_fib_dump_reg(void)
+{
+	return dylib;
```

Comment:
We don't need `get_fib_dump_reg()`, let's just rename variable `dylib` to `dylib_lua_c_funcs` and export `dylib_lua_c_funcs`.